sql-plugin/src/main/scala/com/nvidia/spark/rapids/aggregate.scala

```diff
@@ -427,7 +433,10 @@ class GpuHashAggregateIterator(
       batches: mutable.ArrayBuffer[LazySpillableColumnarBatch]): LazySpillableColumnarBatch = {
     withResource(batches) { _ =>
       withResource(concatenateBatches(batches)) { concatVectors =>
-        withResource(computeAggregate(concatVectors, merge = true)) { mergedBatch =>
+        val concatBatch = new ColumnarBatch(
+          concatVectors.toArray,
+          concatVectors.head.getRowCount.toInt)
```

Comment:
Is it possible to have a batch with no rows in it here? I don't think it is, but every time we assume something like this I want to be sure.

Reply:
I don't think we could (i.e a Filter adds a coalesce, and after a shuffle we coalesce also), but I did change this impl a bit.

Reply:
If the input to the agg is a join, I do not know what happens. If the join could output empty batches.